### DISCUSSION TITLE
latest version 0.2.38 have so many errors on get_earnings_dates() return data

### DISCUSSION THREAD

**Discussion by billionpan2020** [ranaroussi/yfinance#1953] [Development]
### Describe bug

I always use yfinance Reported EPS data do stock analysis. I found the incompatible errors in the lastest version of get_earnings_dates() return data. It organize and return the quarterly eps data clean and accurate on the whole in the past, but the behavior of this version changed, and raise some logical errors that I can not handle and filtered with additional code in my software.

### Simple code that reproduces your problem

stock = yf.Ticker('BMO')
quarterly_earnings = stock.get_earnings_dates(limit=40)
quarterly_earnings

### Debug log

There is not a syntax error, it is just a logical issue that should be handled before return the final result of this method. the following  attachment is the return values when I run the above sample code in jupyter notebook. the issue show up clearly after the date 2023-12-01. between the date 2024-01-30 and 2024-04-16, the data is just such a mess.

please see the attachment.
![running result](https://github.com/ranaroussi/yfinance/assets/129495891/aeefb86c-ad90-4a12-8b5b-40dce85b6e5e)

### `yfinance` version

0.2.38

**Comment by ValueRaider** [comment]
Does Yahoo website have different data?

**Comment by StephanAkkerman** [comment]
This is the same data as on the website: https://finance.yahoo.com/calendar/earnings?symbol=BMO

**Comment by billionpan2020** [comment]
ok, that seems it is not a error in function behavior, but that recent data on the website make the usage of it impossible.  
In the past, there is only one eps data each quarter on the whole, however when you look at the recent data after 2024-12-01, you can not know which data belong to which quarter and which data is the right number

**Comment by StephanAkkerman** [comment]
It seems like the bug also occurs for other companies. For instance, NVIDIA also seems to have the wrong data: https://finance.yahoo.com/calendar/earnings?symbol=NVDA

**Comment by billionpan2020** [comment]
What can we do for such data error on Yahoo website?  report to the technical suport teem of Yahoo finance?

**Comment by StephanAkkerman** [comment]
I suppose that would be an option however, it will take time. I think if you want to use their data and have a quick fix you could simply ignore the data where EPS estimate, Reported EPS, and Surprise (%) are the same as the previous report.

**Comment by billionpan2020** [comment]
if their error data just duplicate the previous correct data, I can fix it or just ignore it. however, you will notice that the extra data shown every month changed without a certain pattern and I can not determine those data are the wrong last quarter data or the new quarter data when the next quarter report come

**Comment by StephanAkkerman** [comment]
You're right, I now see that it sometimes deviates for some reason

**Comment by billionpan2020** [comment]
So, should I close this issue now? or just leave it open for future checking on this issue?
